File: app-wear/src/main/java/de/tu_darmstadt/seemoo/nfcgate/wear/model/CaptureState.java
package de.tu_darmstadt.seemoo.nfcgate.wear.model;

public class CaptureState {
    private final int sessionCount;
    private final long bytesCaptured;
    private final boolean active;
    private final boolean paused;

    public CaptureState(int sessionCount, long bytesCaptured, boolean active, boolean paused) {
        this.sessionCount = sessionCount;
        this.bytesCaptured = bytesCaptured;
        this.active = active;
        this.paused = paused;
    }

    public int getSessionCount() {
        return sessionCount;
    }

    public long getBytesCaptured() {
        return bytesCaptured;
    }

    public boolean isActive() {
        return active;
    }

    public boolean isPaused() {
        return paused;
    }
}

File: app-wear/src/main/java/com/nfcgate/wear/model/CaptureState.kt
package com.nfcgate.wear.model

data class CaptureState(
    val active: Boolean = false,
    val paused: Boolean = false,
    val tapCount: Int = 0,
    val bytes: Long = 0L,
    val durationSeconds: Long = 0L,
    val phoneConnected: Boolean = false,
    val batteryPercent: Int = -1,
    val recentTaps: List<TapRecord> = emptyList()
)

data class TapRecord(
    val type: String,
    val data: String,
    val timestamp: Long
)
